### PR TITLE
Created Teledyne base classes with most of `LeCroyT3DSO1204` functionality

### PR DESCRIPTION
**Pull request by RobertoRoos**
This PR is split of from my original: #868 
This in the progress of fixing #857 

Note: I also applied the patches on `LeCroyT3DSO1204` from March until now on the original file and the new Teledyne class. This diff can be viewed at: https://github.com/pymeasure/pymeasure/compare/9afd3c..41675c#diff-cc461e0399252211cf6d632e042fcbf0e4e7ae44b99f9550fad375e8ce360f4b

EDIT: I will work on fixing the docstrings next.

### CODE REVIEW

File: pymeasure/instruments/teledyne/teledyne_oscilloscope.py [hunk 1]
```diff
@@ -0,0 +1,1136 @@
...
+            raise ValueError("LINE is not a valid channel")
+        else:
+            return getattr(self, f"ch_{source if isinstance(source, int) else source[-1]}")
+
+    def autoscale(self):
+        """ Autoscale displayed channels."""
+        self.write("ASET")
+
+    def write(self, command, **kwargs):
+        """ Writes the command to the instrument through the adapter.
```
Comment (bmoneke) [inline pymeasure/instruments/teledyne/teledyne_oscilloscope.py:541]: Just a small thing, as we are touching the whole file:
```suggestion
        """ Write the command to the instrument through the adapter.
```
(docstrings should have their first phrase in imperative mode.)

File: pymeasure/instruments/teledyne/teledyne_oscilloscope.py [hunk 2]
```diff
@@ -0,0 +1,1136 @@
...
+    # The third response item ("state"), is not present in all oscilloscopes
+    # For compatibility it is captured if it can, but ignored otherwise
+    _re_pava_response = re.compile(r"^\s*"
+                                   r"(?P<parameter>\w+),\s*"
+                                   r"(?P<value>[^,]*)\s*"
+                                   r"(?:,(?P<state>\w+)\s*)?$")
+
+    bwlimit = Instrument.control(
+        "BWL?", "BWL %s",
+        """Set the internal low-pass filter for all channels.
```
Comment (bmoneke) [inline pymeasure/instruments/teledyne/teledyne_oscilloscope.py:224]: Is it really all channels? The command includes only a single channel in its name (from `insert_id`) and the instrument has a bwlimit command as well, which sets all channels.

Mabye a wrong copy paste from the wrong position?

**Review by bmoneke — CHANGES_REQUESTED** [review]
A few comments, but I'm confident, we will merge it soon.

As this is your first PR which is going to be merged, you might include the authors.txt here as well.

As you clean the docstrings, you might remove the following lines, which allow the instrument to skip the docstring test.

https://github.com/pymeasure/pymeasure/blob/6ff994ee3c4a74ce641b61b4c955093de490980c/tests/instruments/test_all_instruments.py#L177

https://github.com/pymeasure/pymeasure/blob/6ff994ee3c4a74ce641b61b4c955093de490980c/tests/instruments/test_all_instruments.py#L207

File: pymeasure/instruments/lecroy/lecroyT3DSO1204.py [hunk 3]
```diff
@@ -163,351 +67,74 @@ def _measure_delay_validator(value, values):
...
+    """
 
-    Implementation modeled on Channel object of Keysight DSOX1102G instrument. """
+    TRIGGER_SLOPES = {"negative": "NEG", "positive": "POS", "window": "WINDOW"}
 
-    _BOOLS = {True: "ON", False: "OFF"}
+    # Change listed values for existing commands:
+    trigger_slope_values = TRIGGER_SLOPES
 
     bwlimit = Instrument.control(
```
[the remark this answers is not in the document]
Comment (RobertoRoos) [reply pymeasure/instruments/lecroy/lecroyT3DSO1204.py:81]: Yeah, but I think the options were different for different oscilloscopes. I emphasized this in the doc.

File: pymeasure/instruments/teledyne/teledyne_oscilloscope.py [hunk 4]
```diff
@@ -0,0 +1,1136 @@
...
+            "trigger_slope": self.trigger_slope
+        }
+        return ch_setup
+
+
+class TeledyneOscilloscope(Instrument, metaclass=ABCMeta):
+    """A base abstract class for any Teledyne Lecroy oscilloscope.
+
+    All Teledyne oscilloscopes have a very similar interface, hence this basic base class to combine
+    them.
```
[the remark this answers is not in the document]
Comment (RobertoRoos) [reply pymeasure/instruments/teledyne/teledyne_oscilloscope.py]: Very basic base :p

**Comment by RobertoRoos** [comment]
Oops, I had already forgotten about this...
@bmoneke can we merge?

**Comment by bmoneke** [comment]
Sorry, I was on vacation.

**Pull request merged**